lib/Collection.js

```diff
@@ -94,6 +94,22 @@ module.exports = class Collection {
   }
 
   async findDocuments({id, type, equals, has}) {
+    equals = this._prepareQuery({id, type, equals, has});
+    const {capability, edvClient, invocationSigner} = this;
+    const results = await edvClient.find(
+      {equals, has, capability, invocationSigner});
+    return results;
+  }
+
+  async countDocuments({id, type, equals, has}) {
+    equals = this._prepareQuery({id, type, equals, has});
+    const {capability, edvClient, invocationSigner} = this;
+    const documentCount = await edvClient.count(
+      {equals, has, capability, invocationSigner});
+    return documentCount;
+  }
+
+  _prepareQuery({id, type, equals, has}) {
```

Comment:
Since this is only returning `equals` (it doesn't cover `has` which is also part of the query) let's put `equals` in the name.

```suggestion
  _prepareEqualsQuery({id, type, equals, has}) {
```

Reply:
Does `has` stay as a parameter?  Is it not being used at all?

Reply:
there is a check in the ` _prepareEqualsQuery`  function that throws error if any of the parameters including `has` is not given. That's the only place it's being used inside that function.

Reply:
hmmm.. not liking the sound of that, maybe the private function is doing too much if it's also doing validation?

Reply:
I can remove `has` from the private function, and do the validation separately inside `countDocuments` and `findDocuments` methods... That way the private function will only concern about equals.

Reply:
Sounds great!